test.py

```diff
@@ -121,17 +125,40 @@ def transfer_token(client, recipient_id, token_id):
         print(f"Token transfer failed: {str(e)}")
         sys.exit(1)
 
+def delete_token(client, token_id, admin_key):
+    """Deletes the specified token on the Hedera network."""
+    transaction = (
+        TokenDeleteTransaction()
+        .set_token_id(token_id)
+        .freeze_with(client)
+    )
+    
+    transaction.sign(client.operator_private_key)
+    transaction.sign(admin_key)
+
+    try:
+        receipt = transaction.execute(client)
+        if receipt.status != ResponseCode.SUCCESS:
+            status_message = ResponseCode.get_name(receipt.status)
+            raise Exception(f"Token deletion failed with status: {status_message}")
+        print("Token deletion successful.")
+    except Exception as e:
+        print(f"Token deletion failed: {str(e)}")
+        sys.exit(1)
+
 def main():
     operator_id, operator_key = load_operator_credentials()
+    admin_key = PrivateKey.generate()
 
     network = Network(node_address='localhost:50211', node_account_id=AccountId(0, 0, 3))
```

Comment:
rebase please as this has changed (Network env var now comes from either solo pipeline run or local .env file so it's possible to test other networks without running solo :) )